src/languages/groovy.js

```diff
@@ -5,69 +5,84 @@
  Website: https://groovy-lang.org
  */
 
+import * as regex from "../lib/regex";
+
+function variants(variants, obj = {}) {
+  obj.variants = variants;
+  return obj;
+}
+
 export default function(hljs) {
+  const IDENT_RE = '[A-Za-z0-9_$]+';
+  const COMMENT = variants([
+    hljs.C_LINE_COMMENT_MODE,
+    hljs.C_BLOCK_COMMENT_MODE,
+    hljs.COMMENT(
+      '/\\*\\*',
+      '\\*/',
+      {
+        relevance : 0,
+        contains : [
+          {
+            // eat up @'s in emails to prevent them to be recognized as doctags
+            begin: /\w+@/, relevance: 0
+          }, {
+            className : 'doctag',
+            begin : '@[A-Za-z]+'
+          }
+        ]
+      }
+    )
+  ]);
+  const REGEXP = {
+    className: 'regexp',
+    begin: /~?\/[^\/\n]+\//,
+    contains: [
+      hljs.BACKSLASH_ESCAPE
+    ]
+  };
+  const NUMBER = variants([
+    hljs.BINARY_NUMBER_MODE,
+    hljs.C_NUMBER_MODE,
+  ]);
+  const STRING = variants([
+    {
+      begin: /"""/,
+      end: /"""/
+    }, {
+      begin: /'''/,
+      end: /'''/
+    }, {
+      begin: "\\$/",
+      end: "/\\$",
+      relevance: 10
+    },
+    hljs.APOS_STRING_MODE,
+    hljs.QUOTE_STRING_MODE,
+    ],
+    { className: "string" }
+  );
+
     return {
         name: 'Groovy',
         keywords: {
-            literal : 'true false null',
+            built_in: 'this super',
```

Comment:
What is the reason to make these built-ins? I looked into https://docs.groovy-lang.org/latest/html/documentation/ and found them among keywords.

Reply:
They are still keywords, everything in `keywords` is keywords, the question is what kind of keyword on they.  These are very clearly not language keywords in the same vein as `for`, `if`, etc... they are built in variables, same as `global` or `window`, etc...